### ISSUE TITLE
Feature request: Handle missing axis and buttons gracefully

### ISSUE THREAD

**Issue by Petrox** [ros-teleop/teleop_tools#17]
Hi Bence,

When using a joystick with less number of axis than configured in the yaml, joy_teleop drops some exceptions like

```
[ERROR] [WallTime: 1458832204.233291] bad callback: <bound method JoyTeleop.joy_callback of <__main__.JoyTeleop instance at 0x7fb0c71aa830>>
Traceback (most recent call last):
  File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/topics.py", line 720, in _invoke_callback
    cb(msg)
  File "/home/rescube/r16_ws/src/teleop_tools/joy_teleop/scripts/joy_teleop.py", line 66, in joy_callback
    self.run_command(c, data)
  File "/home/rescube/r16_ws/src/teleop_tools/joy_teleop/scripts/joy_teleop.py", line 157, in run_command
    self.run_topic(command, joy_state)
  File "/home/rescube/r16_ws/src/teleop_tools/joy_teleop/scripts/joy_teleop.py", line 184, in run_topic
    val = joy_state.axes[mapping['axis']] * mapping.get('scale', 1.0) + mapping.get('offset', 0.0)
IndexError: tuple index out of range
```

 I believe handling the situation with a message like "Joystick has no axis #3, check your configuration, or input device." would make joy_teleop user friendlier.

 :)

**Comment by bmagyar** [comment]
Nice catch!

For mere documentation purposes could you please drop here the relevant block from your yaml file and a line of `rostopic pub /joy`?

**Comment by Petrox** [comment]
yaml:

```
teleop:
  move:
    type: topic
    message_type: geometry_msgs/Twist
    topic_name: cmd_vel
    axis_mappings:
      -
        axis: 1
        target: linear.x
        scale: 1
      -
        axis: 0
        target: angular.z
        scale: 3
```

rostopic echo /joy

```
header: 
  seq: 188
  stamp: 
    secs: 1458835957
    nsecs: 662428341
  frame_id: ''
axes: [0.009594055823981762, -0.32171809673309326]
buttons: [0]
---
```

**Comment by bmagyar** [comment]
Thanks, this has been fixed by #18
